server/workflowarchive/archived_workflow_server.go

```diff
@@ -40,8 +40,12 @@ func NewWorkflowArchiveServer(wfArchive sqldb.WorkflowArchive, offloadNodeStatus
 }
 
 func (w *archivedWorkflowServer) ListArchivedWorkflows(ctx context.Context, req *workflowarchivepkg.ListArchivedWorkflowsRequest) (*wfv1.WorkflowList, error) {
+	listOption := metav1.ListOptions{}
+	if req.ListOptions != nil {
+		listOption = *req.ListOptions
+	}
```

Comment:
looks like this was the missing piece from https://github.com/argoproj/argo-workflows/pull/12736#discussion_r1815439059

```suggestion
	listOptions := metav1.ListOptions{}
	if req.ListOptions != nil {
		listOptions = *req.ListOptions
	}
```

pluralize

Reply:
actually, this might make sense to move to the `BuildListOptions` helper and have it able to handle the possibility of `nil`

Reply:
updated